lib/carrier/messaging/tracker.ex

```diff
@@ -0,0 +1,179 @@
+defmodule Carrier.Messaging.Tracker do
+
+  defstruct [reply_endpoints: %{}, subscriptions: %{}, monitors: %{}, unused_topics: []]
+
+  @spec add_reply_endpoint(Tracker.t, pid()) :: {Tracker.t, String.t}
+  def add_reply_endpoint(%__MODULE__{reply_endpoints: reps}=tracker, subscriber) do
+    case Map.get(reps, subscriber) do
+      nil ->
+        topic = make_reply_endpoint_topic()
+        reps = reps
+        |> Map.put(subscriber, topic)
+        |> Map.put(topic, subscriber)
```

Comment:
So `reply_endpoints` is actually intended to serve as a bi-directional lookup linking each subscriber process to their unique message queue topic (and vice versa), correct?

If so, is there a better name that could capture that? `reply_endpoints` sounds like it'd be a list of topics.